common/djangoapps/embargo/middleware.py

```diff
@@ -73,27 +73,62 @@ def __init__(self):
         # If embargoing is turned off, make this middleware do nothing
         if not settings.FEATURES.get('EMBARGO', False) and not self.site_enabled:
             raise MiddlewareNotUsed()
+        self.new_implementation_enabled = settings.FEATURES.get('NEW_MIDDLE_WARE_IMPLEMENTATION', False)
 
     def process_request(self, request):
         """
         Processes embargo requests.
         """
+        if self.new_implementation_enabled:
+            return self.new_implementation(request)
+        else:
+            url = request.path
+            course_id = course_id_from_url(url)
+            course_is_embargoed = EmbargoedCourse.is_embargoed(course_id)
+
+            # If they're trying to access a course that cares about embargoes
+            if self.site_enabled or course_is_embargoed:
+
+                # Construct the list of functions that check whether the user is embargoed.
+                # We wrap each of these functions in a decorator that logs the reason the user
+                # was blocked.
+                # Each function should return `True` iff the user is blocked by an embargo.
+                check_functions = [
+                    self._log_embargo_reason(check_func, course_id, course_is_embargoed)
+                    for check_func in [
+                        partial(self._is_embargoed_by_ip, get_ip(request)),
+                        partial(self._is_embargoed_by_profile_country, request.user)
+                    ]
+                ]
+
+                # Perform each of the checks
+                # If the user fails any of the checks, immediately redirect them
+                # and skip later checks.
+                for check_func in check_functions:
+                    if check_func():
+                        return self._embargo_redirect_response
+
+            # If all the check functions pass, implicitly return None
+            # so that the middleware processor can continue processing
+            # the response.
+
+    def new_implementation(self, request):
         url = request.path
         course_id = course_id_from_url(url)
-        course_is_embargoed = EmbargoedCourse.is_embargoed(course_id)
+        course_is_restricted = RestrictedCourse().is_restricted_course(course_id)
 
         # If they're trying to access a course that cares about embargoes
-        if self.site_enabled or course_is_embargoed:
+        if self.site_enabled or course_is_restricted:
```

Comment:
We don't need to check `self.site_enabled` anymore.  As far as I can tell, the feature flag it uses (`SITE_EMBARGOED`) was never actually used in production, so I'm hoping we can remove it in our new implementation.